logrusx/helper.go

```diff
@@ -98,6 +99,61 @@ func (l *Logger) WithRequest(r *http.Request) *Logger {
 	return ll
 }
 
+func (l *Logger) Logf(level logrus.Level, format string, args ...interface{}) {
+	if !l.leakSensitive {
+		var myArgs []interface{}
+		for _, arg := range args {
+			urlArg, ok := arg.(*url.URL)
+			if ok {
+				urlCopy := url.URL{Scheme: urlArg.Scheme, Host: urlArg.Host, Path: urlArg.Path}
+				myArgs = append(myArgs, &urlCopy)
+			} else {
+				myArgs = append(myArgs, arg)
+			}
+		}
```

Comment:
One last optimization we can do, just FYI:
```suggestion
		for i, arg := range args {
			urlArg, ok := arg.(*url.URL)
			if ok {
				urlCopy := url.URL{Scheme: urlArg.Scheme, Host: urlArg.Host, Path: urlArg.Path}
				arg[i] = &urlCopy
			}
		}
```

Reply:
And maybe even support both `url.URL` and `*url.URL` for the sake of completeness:
```suggestion
		for i, arg := range args {
			switch urlArg := arg.(type) {
			case url.URL:
				urlCopy := url.URL{Scheme: urlArg.Scheme, Host: urlArg.Host, Path: urlArg.Path}
				arg[i] = urlCopy
			case *url.URL:
				urlCopy := url.URL{Scheme: urlArg.Scheme, Host: urlArg.Host, Path: urlArg.Path}
				arg[i] = &urlCopy
			default:
				continue
			}
		}
```